sdk/trace/span.go

```diff
@@ -286,3 +304,43 @@ func startSpanInternal(name string, parent core.SpanContext, remoteParent bool,
 
 	return span
 }
+
+type samplingDecision struct {
+	noParent     bool
+	remoteParent bool
+}
+
+type samplingData struct {
+	parent core.SpanContext
+	name   string
+	cfg    *Config
+	span   *span
+}
+
+type makeDecisionData func() samplingData
```

Comment:
Ca we just pass `samplingData` instead of `func() samplingData`?

Reply:
Sure, can change that. But the reason for passing a function producing samplingData instead of passing the sampling data itself was that the sampling data might not be used at all if the span has a local parent.